W-21217280 mcp creator gr (#639)
mulesoft/docs-api-manager: modules/ROOT/pages/create-instance-task-mcp-bridge.adoc

= Adding an MCP Bridge Instance
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:product: flex

Flex Gateway MCP Bridge enables you to create MCP servers from your existing API instances. With the MCP Bridge, you can expose your current API instances as MCP tools so agents can discover and call them.

> Define your MCP bridge by selecting API instances and mapping them to MCP tools. When an agent requests the list of available tools (`tools/list`),the gateway responds using an MCP descriptor that defines each tool's name, description, and input schema. When an agent calls a tool (`tools/call`), the MCP Bridge maps the tool name and arguments to an HTTP request based on your configuration.

Reviewer: ```suggestion
Define your MCP bridge by selecting API instances and mapping them to MCP tools. When an agent requests the list of available tools (`tools/list`), the gateway responds by using an MCP descriptor that defines each tool's name, description, and input schema. When an agent calls a tool (`tools/call`), the MCP Bridge maps the tool name and arguments to an HTTP request based on your configuration.
```

This diagram shows the relationship of the upstream and downstream configurations to your upstream service and agent client:

> image:mcp-bridge-instance.png[The MCP Bridge instance is deployed on a gateway between the upstream apis and downstream configuration]

Reviewer: ```suggestion
image:mcp-bridge-instance.png["The MCP Bridge instance is deployed on a gateway between the upstream apis and downstream configuration"]
```

NOTE: MCP Bridge doesn't support API instances with multiple upstream services.

== Before You Begin

[[add-server]]
== Add a New Agent or Tool Instance

To add and a new MCP Bridge for Flex Gateway:

. Ensure that all API instances you want to add to your MCP Server are running on the same Flex Gateway you want to add your MCP Server to.
. To begin creating your MCP Server instance, you can start from these locations:
** From *API Instances*:

> ... Click on an API instance you want to add to an MCP Server.

Reviewer: ```suggestion
... Click an API instance you want to add to an MCP Server.
```

... Click *Actions* > *Add to new MCP Server instance*.

** From *Agent and Tool Instances*:
... Click *Add* > *MCP Bridge*.

> . Select a Flex Gateway to deploy the MCP server instance to from *Select a gateway*.

Reviewer: ```suggestion
. From *Select a gateway*, choose a Flex Gateway for deploying the MCP server instance.
```

. Click *Next*.
. Select the API instances you want to add to your MCP Server.

> . Define a *MCP asset name*.

Reviewer: ```suggestion
. Define the MCP server's asset name.
```

. Click *Next*.
. Configure the downstream configuration settings:
+
.Managed Flex Gateway
[%collapsible]
====
include::partial$api-configuration-tables.adoc[tags=flex-downstream-agent-tool-managed]
====
+
.Self-Managed Flex Gateway (Connected Mode)
[%collapsible]
====
include::partial$api-configuration-tables.adoc[tags=flex-downstream-agent-tool-self-managed]
====

. Click *Next*.
. Configure the upstream services for your API instances:
+
|===
| Field Name | Description | Required | Notes

> | *Route Label* | Label to better identify multiple API instances. | Yes | By default, the label if the API name.

Reviewer: ```suggestion
| *Route Label* | Label to better identify multiple API instances. | Yes | By default, the label is the API name.
```

> | *Upstream URL* | URL to access for the upstream service. This must end with a `/`. | Yes |  For example, you can use the URL of your asset in Exchange.

Reviewer: ```suggestion
| *Upstream URL* | URL to access for the upstream service. This must end with a `/`. | Yes |  For example, use the URL of your asset in Exchange.
```

| *Outbound TLS* | TLS Context used for the outbound traffic to the upstream service | No | xref:gateway::flex-conn-tls-config.adoc[Configure a TLS Context for Flex Gateway] before adding a TLS Context to your server.
|===
+

> To switch amoung API instances, select the name of the API.

Reviewer: ```suggestion
To switch among API instances, select the name of the API.
```
